### PR TITLE
Case flags data migration

### PR DESCRIPTION
**Pull request by Shailesh-walke**
### Change description ###

Enter a description.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SPTRIBS-

**Before merging a pull request make sure that:**

- [ ] tests have been updated / new tests has been added (if needed)
- [ ] README and other documentation has been updated / added (if needed)

**If this ticket will have any visible impact on users and is not behind a feature toggle, make sure that:**
- [ ] this ticket been reviewed by QA
- [ ] the user story been signed off by the PO

Note that bug fixes, dependency updates and technical tasks do not directly impact the user experience and can be merged without QA and PO review.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -326,7 +326,9 @@ def sonarExclusions = [
   '**/uk/gov/hmcts/sptribs/common/event/CreateTestCase.java',
   '**/uk/gov/hmcts/sptribs/CaseApiApplication.java',
   '**/uk/gov/hmcts/sptribs/notification/GetAllTemplatesFromNotify.java',
-  '**/uk/gov/hmcts/sptribs/common/config/**'
+  '**/uk/gov/hmcts/sptribs/common/config/**',
+  '**/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java',
```
Comment (tomxelliott) [inline build.gradle:330]: Please ensure these exclusions are removed after testing

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlags.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+
+import static uk.gov.hmcts.sptribs.ciccase.model.UserRole.SYSTEMUPDATE;
+import static uk.gov.hmcts.sptribs.ciccase.model.access.Permissions.CREATE_READ_UPDATE_DELETE;
+
+@Component
+@Slf4j
+public class SystemMigrateCaseFlags implements CCDConfig<CaseData, State, UserRole> {
+
+    public static final String SYSTEM_MIGRATE_CASE_FLAGS = "system-migrate-case-flags";
+    //public static final String SYSTEM_MIGRATE_CASE_FLAGS = "system-migrate-old-hearing";
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlags.java:26]: can this not be removed?

File: build.gradle (hunk 1 above)
Comment (Shailesh-walke) [reply build.gradle]: removed them

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlags.java (hunk 2 above)
Comment (Shailesh-walke) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/event/SystemMigrateCaseFlags.java]: Removed

**Review by sjamihmcts — APPROVED** [review]

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        log.info("Migrate case flags scheduled task started");
+
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query =
+                boolQuery()
+                    .must(boolQuery()
+                        .mustNot(existsQuery("data.subjectFlags"))
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java:52]: Are you not going to test on a single case first to verify before merging this query?

**Review by tomxelliott — CHANGES_REQUESTED** [review]
as per previous discussion I believe there was going to be a sanity check on a single case before migrating all existing cases

**Comment by Shailesh-walke** [comment]
> as per previous discussion I believe there was going to be a sanity check on a single case before migrating all existing cases

updated query for pulling single record

File: src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        final User user = idamService.retrieveSystemUpdateUserDetails();
+        final String serviceAuth = authTokenGenerator.generate();
+
+        try {
+            final BoolQueryBuilder query =
+                boolQuery()
+                    .must(boolQuery()
+                        .must(matchQuery("reference", 1688978122333564L))
+                    )
+                ; //temporary query change to test for single record
```
Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java:54]: semi colon shouldn't be on a new line

Comment (tomxelliott) [inline src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java:54]: Mark this comment with a TODO so it isn't missed

Comment (Shailesh-walke) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java]: updated

Comment (Shailesh-walke) [reply src/main/java/uk/gov/hmcts/sptribs/systemupdate/schedule/migration/SystemMigrateCaseFlagsTask.java]: updated with TODO

**Review by sjamihmcts — APPROVED** [review]

**Review by tomxelliott — APPROVED** [review]

**Pull request merged**